sources/packages/api/src/services/application/application.service.ts

```diff
@@ -1288,4 +1297,71 @@ export class ApplicationService extends RecordDataModelService<Application> {
     }
     return applicationQuery.getOne();
   }
+
+  /**
+   * Validation for application overlapping dates or Pending PIR.
+   * This validation can be disabled by setting BYPASS_APPLICATION_SUBMIT_VALIDATIONS to true in .env file.
+   * @param applicationId
+   * @param lastName
+   * @param userId
+   * @param sin
+   * @param birthDate
+   * @param studyStartDate
+   * @param studyEndDate
+   */
+  async validateOverlappingDatesAndPIR(
+    applicationId: number,
+    lastName: string,
+    userId: number,
+    sin: string,
+    birthDate: Date,
+    studyStartDate: Date,
+    studyEndDate: Date,
+  ): Promise<void> {
+    if (!this.config.bypassApplicationSubmitValidations) {
+      const existingOverlapApplication = this.validatePIRAndDateOverlap(
+        userId,
+        studyStartDate,
+        studyEndDate,
+        applicationId,
+      );
+
+      const existingSFASFTApplication =
+        this.sfasApplicationService.validateDateOverlap(
+          sin,
+          birthDate,
+          lastName,
+          studyStartDate,
+          studyEndDate,
+        );
+
+      const existingSFASPTApplication =
+        this.sfasPartTimeApplicationsService.validateDateOverlap(
+          sin,
+          birthDate,
+          lastName,
+          studyStartDate,
+          studyEndDate,
+        );
+      const [
+        applicationResponse,
+        sfasFTApplicationResponse,
+        sfasPTApplicationResponse,
+      ] = await Promise.all([
+        existingOverlapApplication,
+        existingSFASFTApplication,
+        existingSFASPTApplication,
+      ]);
+      if (
+        !!applicationResponse ||
+        !!sfasFTApplicationResponse ||
+        !!sfasPTApplicationResponse
+      ) {
+        throw new CustomNamedError(
+          PIR_OR_DATE_OVERLAP_ERROR_MESSAGE,
+          INVALID_OPERATION_IN_THE_CURRENT_STATUS,
```

Comment:
I believe that PIR_OR_DATE_OVERLAP_ERROR would be better than INVALID_OPERATION_IN_THE_CURRENT_STATUS, right?

Reply:
Actually, even `PIR_OR_DATE_OVERLAP_ERROR ` does not seem right 😄 
This would be simple APPLICATION_DATE_OVERLAP_ERROR, IMO.